src/fw/applib/ui/kino/kino_layer.c

```diff
@@ -35,7 +72,17 @@ static void prv_update_proc(Layer *layer, GContext *ctx) {
   }
 
   const GRect reel_bounds = kino_layer_get_reel_bounds(kino_layer);
-  kino_player_draw(&kino_layer->player, ctx, reel_bounds.origin);
+  // Check the inversion flag stored in the context pointer's LSB
+  if (KINO_LAYER_GET_INVERT_FLAG(kino_layer)) {
+    GDrawCommandProcessor gdraw_inv_processor = {
+        .command = prv_invert_pdc_colors,
+    };
+
+    KinoReelProcessor invert_processor = {.draw_command_processor = &gdraw_inv_processor};
+    kino_player_draw_processed(&kino_layer->player, ctx, reel_bounds.origin, &invert_processor);
+  } else {
+    kino_player_draw(&kino_layer->player, ctx, reel_bounds.origin);
```

Comment:
Considering that the inverter processor doesn't require any local state and that `kino_reel_draw` ultimately calls `kino_reel_draw_processed` with a `NULL` processor anyway, maybe there's a chance to simplify this section of the function to something like

```
KinoReelProcessor processor =
  kino_layer->invert_colors ?
  PRV_INVERT_COLORS_PROCESSOR :
  NULL;
kino_player_draw_processed(..., & processor);
```